src/lua_console.cpp

```diff
@@ -0,0 +1,82 @@
+#include "lua_console.h"
+
+#include "catalua.h"
+#include "catacharset.h"
+
+lua_console::lua_console() : cWin( newwin( lines, width, 0, 0 ) ), iWin( newwin( 1, width,lines, 0 ) )
+{
+}
+
+void lua_console::dispose()
+{
+    werase( cWin );
+    werase( iWin );
+    delwin( cWin );
+    delwin( iWin );
+}
+
+std::string lua_console::get_input()
+{
+    long key = 0;
+    int pos = -1;
+    return string_input_win( iWin, "", width, 0, 0, width, true, key, pos, "LUA", 0, lines, true, false, callbacks );
+}
+
+void lua_console::draw()
+{
+    werase( cWin );
+
+    // Some juggling to make sure text is aligned with the bottom of the console.
+    int stack_size = text_stack.size() - scroll;
```

Comment:
I looked into using ```fold_and_print_from``` and there's 2 problems with it. 

First, I still can't align the text with the bottom of the console, I would have to know how many lines it prints before calling, otherwise the wrapped part would just overflow the window and be truncated.

Second, I'm printing the lines in reversed order, so the newest one is at the bottom. That way the window automatically scrolls (sorta). Using ```fold_and_print_from``` would mean you would have to manually scroll down to the new output as soon as the number of lines exceeds 10. This problem could be fixed by storing the lines in reversed order, but the lines that get folded would have to be cut up first so they don't appear the wrong way around. This isn't very different from what I have now, besides, constantly resizing a string isn't very efficient either, I think using the vector is the better solution.